internal/logger/default_logger.go

```
@@ -21,124 +21,122 @@ import (
+// New returns a Default LogAdaptor with defaultLogLevel.
 func New() *DefaultLogger {
 	return &DefaultLogger{
 		Logger: log.New(os.Stderr, "", log.LstdFlags),
 		Level:  defaultLogLevel,
 	}
 }
 
-func NewWithLevel(loggingLevel int) *DefaultLogger {
+func NewWithLevel(loggingLevel logger.Level) *DefaultLogger {
 	return &DefaultLogger{
 		Logger: log.New(os.Stderr, "", log.LstdFlags),
 		Level:  loggingLevel,
 	}
 }
 
-// Debug logs the given arguments at debug level if the level is greater than or equal to debug level.
-func (l *DefaultLogger) Debug(f func() string) {
-	if l.CanLogDebug() && f != nil {
-		s := fmt.Sprintf("DEBUG: %s", f())
-		l.Output(logCallDepth, s)
-	}
-}
-
-func (l *DefaultLogger) Trace(f func() string) {
-	if l.canLogTrace() && f != nil {
-		s := fmt.Sprintf("TRACE: %s", f())
-		l.Output(logCallDepth, s)
+func (l *DefaultLogger) canLog(level logger.Level) bool {
+	numericLevel, err := logger.GetLogLevel(level)
+	if err != nil {
+		return false
 	}
-}
-
-// Info logs the given arguments at info level if the level is greater than or equal to info level.
-func (l *DefaultLogger) Info(args ...interface{}) {
-	if l.canLogInfo() {
-		callerName := l.findCallerFuncName()
-		s := callerName + "\n" + infoPrefix + ": " + fmt.Sprint(args...)
-		l.Output(logCallDepth, s)
+	loggerLevel, err := logger.GetLogLevel(l.Level)
+	if err != nil {
+		fmt.Println("logger has invalid logger level, will not logger something useful")
+		return false
 	}
+	return loggerLevel >= numericLevel
 }
 
-func (l *DefaultLogger) Infof(format string, values ...interface{}) {
-	if l.canLogInfo() {
-		s := fmt.Sprintf("INFO : %s", fmt.Sprintf(format, values...))
-		l.Output(logCallDepth, s)
+func (l *DefaultLogger) Log(level logger.Level, formatter func() string) {
+	if !l.canLog(level) || formatter == nil {
+		return
 	}
+	s := fmt.Sprintf("%s: %s", strings.ToUpper(level.String()), formatter())
+	_ = l.Output(logCallDepth, s)
```

How about a comment explaining why the error was ignored?

---

Sure, just thought it is common practice to ignore logging error. Added a comment as suggested